### ISSUE TITLE
TypeDeclarationFromSuperclassExtractor.getCompilationUnit() throws NoSuchMethodError with Eclipse Neon (4.6.3)

### ISSUE THREAD

**Issue by andi-livn** [helospark/SparkBuilderGenerator#46]
This issue was not present prior to version 0.0.19's fix for #44.
On my admittedly older Eclipse version 4.6.3 (Neon.3) invoking the generator now fails with the following error in the Eclipse logs:

```
!ENTRY org.eclipse.ui 4 0 2018-11-26 15:37:38.400
!MESSAGE Unhandled event loop exception
!STACK 0
java.lang.NoSuchMethodError: org.eclipse.jdt.core.IType.getClassFile()Lorg/eclipse/jdt/core/IOrdinaryClassFile;
	at com.helospark.spark.builder.handlers.codegenerator.component.helper.TypeDeclarationFromSuperclassExtractor.getCompilationUnit(TypeDeclarationFromSuperclassExtractor.java:65)
	at com.helospark.spark.builder.handlers.codegenerator.component.helper.TypeDeclarationFromSuperclassExtractor.extractTypeDeclaration(TypeDeclarationFromSuperclassExtractor.java:45)
	at com.helospark.spark.builder.handlers.codegenerator.component.helper.TypeDeclarationFromSuperclassExtractor.lambda$0(TypeDeclarationFromSuperclassExtractor.java:37)
	at java.util.Optional.flatMap(Unknown Source)
	at com.helospark.spark.builder.handlers.codegenerator.component.helper.TypeDeclarationFromSuperclassExtractor.extractTypeDeclarationFromSuperClass(TypeDeclarationFromSuperclassExtractor.java:37)
	at com.helospark.spark.builder.handlers.codegenerator.builderfieldcollector.SuperConstructorParameterCollector.collect(SuperConstructorParameterCollector.java:46)
	at com.helospark.spark.builder.handlers.codegenerator.ApplicableBuilderFieldExtractor.lambda$0(ApplicableBuilderFieldExtractor.java:28)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.ReferencePipeline$Head.forEach(Unknown Source)
	at com.helospark.spark.builder.handlers.codegenerator.ApplicableBuilderFieldExtractor.findBuilderFields(ApplicableBuilderFieldExtractor.java:27)
	at com.helospark.spark.builder.handlers.codegenerator.RegularBuilderCompilationUnitGeneratorBuilderFieldCollectingDecorator.generateBuilder(RegularBuilderCompilationUnitGeneratorBuilderFieldCollectingDecorator.java:35)
	at com.helospark.spark.builder.handlers.GenerateBuilderExecutorImpl.addBuilder(GenerateBuilderExecutorImpl.java:68)
	at com.helospark.spark.builder.handlers.GenerateBuilderExecutorImpl.execute(GenerateBuilderExecutorImpl.java:51)
	at com.helospark.spark.builder.handlers.GenerateBuilderHandlerErrorHandlerDecorator.execute(GenerateBuilderHandlerErrorHandlerDecorator.java:25)
	at com.helospark.spark.builder.handlers.StateInitializerGenerateBuilderExecutorDecorator.execute(StateInitializerGenerateBuilderExecutorDecorator.java:26)
	at com.helospark.spark.builder.handlers.GenerateDefaultBuilderHandler.execute(GenerateDefaultBuilderHandler.java:33)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:295)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:90)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:55)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:282)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:264)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:132)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:152)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:494)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:488)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:210)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:433)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:454)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem$3.handleEvent(AbstractContributionItem.java:482)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4418)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1079)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4236)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3824)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1121)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1022)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:150)
	at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:693)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:610)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:138)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1519)
```

**Comment by deepakab03** [comment]
Same issue in Eclipse Oxygen: 
 Version: Oxygen.3a Release (4.7.3a)
 Build id: 20180405-1200

From Error Log:
_eclipse.buildId=4.7.3.M20180330-0640
java.version=1.8.0_77
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en_US
Framework arguments:  -product org.eclipse.epp.package.jee.product -product org.eclipse.epp.package.jee.product
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.eclipse.epp.package.jee.product -data file:/C:/dev/eclipse-ws/simplified_symmwin/ -product org.eclipse.epp.package.jee.product_

org.eclipse.ui
Error
Mon Nov 26 15:20:03 IST 2018
Unhandled event loop exception

> java.lang.NoSuchMethodError: org.eclipse.jdt.core.IType.getClassFile()Lorg/eclipse/jdt/core/IOrdinaryClassFile;
>     at com.helospark.spark.builder.handlers.codegenerator.component.helper.TypeDeclarationFromSuperclassExtractor.getCompilationUnit(TypeDeclarationFromSuperclassExtractor.java:65)
>     at com.helospark.spark.builder.handlers.codegenerator.component.helper.TypeDeclarationFromSuperclassExtractor.extractTypeDeclaration(TypeDeclarationFromSuperclassExtractor.java:45)
>     at com.helospark.spark.builder.handlers.codegenerator.component.helper.TypeDeclarationFromSuperclassExtractor.lambda$0(TypeDeclarationFromSuperclassExtractor.java:37)
>     at java.util.Optional.flatMap(Optional.java:241)
>     at com.helospark.spark.builder.handlers.codegenerator.component.helper.TypeDeclarationFromSuperclassExtractor.extractTypeDeclarationFromSuperClass(TypeDeclarationFromSuperclassExtractor.java:37)
>     at com.helospark.spark.builder.handlers.codegenerator.builderfieldcollector.SuperConstructorParameterCollector.collect(SuperConstructorParameterCollector.java:46)
>     at com.helospark.spark.builder.handlers.codegenerator.ApplicableBuilderFieldExtractor.lambda$0(ApplicableBuilderFieldExtractor.java:28)
>     at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
>     at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
>     at com.helospark.spark.builder.handlers.codegenerator.ApplicableBuilderFieldExtractor.findBuilderFields(ApplicableBuilderFieldExtractor.java:27)
>     at com.helospark.spark.builder.handlers.codegenerator.RegularBuilderCompilationUnitGeneratorBuilderFieldCollectingDecorator.generateBuilder(RegularBuilderCompilationUnitGeneratorBuilderFieldCollectingDecorator.java:35)
>     at com.helospark.spark.builder.handlers.GenerateBuilderExecutorImpl.addBuilder(GenerateBuilderExecutorImpl.java:68)
>     at com.helospark.spark.builder.handlers.GenerateBuilderExecutorImpl.execute(GenerateBuilderExecutorImpl.java:51)
>     at com.helospark.spark.builder.handlers.GenerateBuilderHandlerErrorHandlerDecorator.execute(GenerateBuilderHandlerErrorHandlerDecorator.java:25)
>     at com.helospark.spark.builder.handlers.StateInitializerGenerateBuilderExecutorDecorator.execute(StateInitializerGenerateBuilderExecutorDecorator.java:26)
>     at com.helospark.spark.builder.handlers.GenerateRegularBuilderHandler.execute(GenerateRegularBuilderHandler.java:26)
>     at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:291)
>     at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:92)
>     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke(Method.java:498)
>     at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:55)
>     at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:305)
>     at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:239)
>     at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:132)
>     at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:152)
>     at org.eclipse.core.commands.Command.executeWithChecks(Command.java:494)
>     at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
>     at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:210)
>     at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:431)
>     at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:446)
>     at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:472)
>     at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
>     at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4428)
>     at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1079)
>     at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4238)
>     at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3817)
>     at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1150)
>     at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
>     at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1039)
>     at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
>     at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:680)
>     at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
>     at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:594)
>     at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
>     at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:151)
>     at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
>     at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
>     at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
>     at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
>     at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
>     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke(Method.java:498)
>     at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
>     at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
>     at org.eclipse.equinox.launcher.Main.run(Main.java:1499)

**Comment by helospark** [comment]
Thanks for letting me know @andi-livn @deepakab03 

It is not caused by the 0.0.19 codechange itself, but issue with compilation.

Eclipse has added [a method](https://github.com/eclipse/eclipse.jdt.core/blame/master/org.eclipse.jdt.core/model/org/eclipse/jdt/core/IType.java#L1241) on the IType interface with covariant return type overriding the one in IMember, compiling with Photon without the correctly defined target platform caused wrong static return type to be included in the generated bytecode, causing the NoSuchMethodError on call.


I'll update the target platform definition and recompile later today.

**Comment by helospark** [comment]
I have released 0.0.20 compiled with the correct target definition.
Please update and check if it works as expected.

**Comment by andi-livn** [comment]
Works as expected in Neon.3. Thanks for the quick fix and this nifty plugin in general.
